src/servant/tt_001/ghost.c

```diff
@@ -258,18 +273,18 @@ Entity* func_us_80171568(Entity* self, s32 entityId) {
 #if defined(VERSION_PSP)
         entity->entityId = entityId;
 #else
-        if (entityId == 0) {
-            entity->entityId = 0xDA;
+        if (entityType == ATTACK_CLOUD) {
+            entity->entityId = ENTITY_ID_ATTACK_CLOUD;
         } else {
-            entity->entityId = 0xDB;
+            entity->entityId = ENTITY_ID_CONFUSION;
         }
 #endif
         entity->zPriority = self->zPriority;
         entity->facingLeft = self->facingLeft;
         entity->flags = FLAG_KEEP_ALIVE_OFFCAMERA;
         entity->posX.val = self->posX.val;
         entity->posY.val = self->posY.val;
-        entity->ext.factory.parent = self;
+        entity->ext.ghostEvent.parent = self;
```

Comment:
I haven't checked this case but a common source of bugs is pointer alignment between parent and child entities

Reply:
I'm not exactly sure what you are meaning, but I added a static assert in entity.h for that field.  If there is something else you'd like me to do, you'll have to let me know.

Reply:
We regularly have an issue where things get misaligned like:

```
addr  | parent obj    | child obj
-----------------------
0x10  | child*        | 
----------------------
0x18  |               | parent*
```

So when the child tries to do self->parent it crashes. Was just giving a warning about it, didn't investigate this case deeply